src/Classes/Utils/ShapeUtils.ts

```diff
@@ -0,0 +1,220 @@
+    }
+
+    private static drawSquareShape(context: CanvasRenderingContext2D, side: number): void {
+        context.rect(-side, -side, side * 2, side * 2);
+    }
+
+    private static drawHeartShape(context: CanvasRenderingContext2D, radius: number): void {
+        const x = -radius / 2;
+        const y = -radius / 2;
+
+        context.moveTo(x, y + radius / 4);
+        context.quadraticCurveTo(x, y, x + radius / 4, y);
+        context.quadraticCurveTo(x + radius / 2, y, x + radius / 2, y + radius / 4);
+        context.quadraticCurveTo(x + radius / 2, y, x + radius * 3 / 4, y);
+        context.quadraticCurveTo(x + radius, y, x + radius, y + radius / 4);
+        context.quadraticCurveTo(x + radius, y + radius / 2, x + radius * 3 / 4, y + radius * 3 / 4);
+        context.lineTo(x + radius / 2, y + radius);
+        context.lineTo(x + radius / 4, y + radius * 3 / 4);
+        context.quadraticCurveTo(x, y + radius / 2, x, y + radius / 4);
+    }
+
+    private static drawTextShape(context: CanvasRenderingContext2D,
+                                 text: string,
+                                 pos: ICoordinates, fill: boolean): void {
+        if (fill) {
+            context.fillText(text, pos.x, pos.y);
+        } else {
+            context.strokeText(text, pos.x, pos.y);
+        }
+    }
+
+    private static drawGenericPolygonShape(context: CanvasRenderingContext2D, start: ICoordinates, side: ISide): void {
+        // By Programming Thomas - https://programmingthomas.wordpress.com/2013/04/03/n-sided-shapes/
+        const sideCount = side.count.numerator * side.count.denominator;
+        const decimalSides = side.count.numerator / side.count.denominator;
+        const interiorAngleDegrees = (180 * (decimalSides - 2)) / decimalSides;
+        const interiorAngle = Math.PI - Math.PI * interiorAngleDegrees / 180; // convert to radians
+
+        if (!context) {
+            return;
+        }
+
+        context.save();
+        context.beginPath();
+        context.translate(start.x, start.y);
+        context.moveTo(0, 0);
+
+        for (let i = 0; i < sideCount; i++) {
+            context.lineTo(side.length, 0);
+            context.translate(side.length, 0);
+            context.rotate(interiorAngle);
+        }
+
+        // c.stroke();
+        context.fill();
+        context.restore();
+    }
+
+    private static drawImageShape(context: CanvasRenderingContext2D, particle: Particle, radius: number): void {
+        if (!context) {
+            return;
+        }
+
+        const imgObj = particle.image?.data.obj;
+
+        if (!imgObj) {
+            return;
+        }
+
+        let ratio = 1;
+
+        if (particle.image) {
+            ratio = particle.image.ratio;
+        }
+
+        const pos = {
+            x: -radius,
+            y: -radius,
+        };
+
+        context.drawImage(imgObj, pos.x, pos.y, radius * 2, radius * 2 / ratio);
+    }
+}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [file should end with a newline](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5171185)